File: apps/atlas-api/src/features/health/domain/health-report.ts
import type { RuntimeContext } from "../../../core/runtime-context";

export type HealthReport = {
  ok: true;
  service: string;
  environment: string;
  destination: string;
  exporter: { endpoint: string; status: "healthy" | "degraded"; signals: string[] };
};

export function toHealthReport(
  ctx: RuntimeContext,
  exporterStatus: "healthy" | "degraded",
): HealthReport {
  return {
    ok: true,
    service: ctx.service,
    environment: ctx.environment,
    destination: ctx.destination,
    exporter: {
      endpoint: ctx.exporterEndpoint,
      status: exporterStatus,
      signals: ["traces", "logs", "metrics"],
    },
  };
}

File: apps/lantern-api/src/features/health/domain/health_report.py
from src.core.runtime_context import RuntimeContext

def to_health(ctx: RuntimeContext, status: str = "healthy") -> dict:
    return {
        "ok": True,
        "service": ctx.service,
        "environment": ctx.environment,
        "destination": ctx.destination,
        "exporter": {
            "endpoint": ctx.exporter_endpoint,
            "status": status,
            "signals": ["traces", "logs", "metrics"],
        },
    }
